model/local.go

```diff
@@ -301,7 +304,16 @@ func (p *LocalModelProvider) QueryText(question string, writer io.Writer, histor
 			return modelResult, nil
 		}
 
-		rawMessages, err := OpenaiGenerateMessages(prompt, question, history, knowledgeMessages, model, maxTokens)
+		const DefaultTokenModel = "gpt-4"
+		var (
+			rawMessages []*RawMessage
+			err         error
+		)
+		if p.subType == "custom-model" {
```

Comment:
You should use this logic:

![8eaf855733ad0c69a59df5caf0746554](https://github.com/user-attachments/assets/f6032223-6d9b-4fc3-aaa3-3c63c7dbc101)



no need to add this code